### ISSUE TITLE
Allow BlankRawContacts to be used in the Delete API

### ISSUE THREAD

**Issue by vestrel00** [vestrel00/contacts-android#220]
## Problem

Thanks to @yuanhoujun in #219, I discovered that the `Delete` API does not support taking in `BlankRawContact`s in the `rawContacts` function.

In order to delete a `BlankRawContact` using the `Delete` API. Users of this library would first have to convert it to a `RawContact`, which requires a database query.

```kotlin
fun deleteBlankRawContacts(context: Context, blankRawContacts: Collection<BlankRawContact>) {
  val contacts = Contacts(context)
  val rawContacts = blankRawContacts.mapNotNull { it.toRawContact(contacts) }
  contacts.delete().rawContacts(rawContacts).commit()
}
```

That is bad for performance!

## Solution

Instead of `Delete.rawContacts(rawContacts: Collection<ExistingRawContactEntity>)`, change it to `Delete.rawContacts(rawContacts: Collection<ExistingRawContactEntityWithContactId>)`.

This fixes the issue because `BlankRawContact` is a subclass of `ExistingRawContactEntityWithContactId` but NOT a subclass of `ExistingRawContactEntity`.

**Comment by vestrel00** [comment]
@yuanhoujun, do you mind if you take on this issue and file a PR for it? I would really love it if you became a contributor! This issue was discovered because of you anyways 😁 

All you would need to do is replace all occurrences of `ExistingRawContactEntity` with `ExistingRawContactEntityWithContactId` in the file [`Delete.kt`](https://github.com/vestrel00/contacts-android/blob/main/core/src/main/java/contacts/core/Delete.kt).

Don't worry about code formatting and documentation if you don't want to deal with those. I can clean up after we merge in your changes.

Feel free to decline this request but it would be great to have another contributor 🙏

**Comment by yuanhoujun** [comment]
Actually, I'm still working on my project "AirController": https://github.com/air-controller/air-controller-desktop.

But I'll try if you need, wait a few minutes.

**Comment by vestrel00** [comment]
> Actually, I'm still working on my project "AirController": https://github.com/air-controller/air-controller-desktop.
> 
> But I'll try if you need, wait a few minutes.

Thanks for doing it anyways! I checked out your GitHub project. Looks good so I gave it a ⭐️ Nice job!

**Comment by vestrel00** [comment]
This is included in release [0.2.2](https://github.com/vestrel00/contacts-android/releases/tag/0.2.2) 🥳
